bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/AggregateDataResponsePublicApi.java

```diff
@@ -12,346 +12,45 @@
  */
 package org.openhab.binding.solaredge.internal.model;
 
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
-import javax.measure.Unit;
-import javax.measure.quantity.Energy;
-
-import org.eclipse.smarthome.core.library.types.DecimalType;
-import org.eclipse.smarthome.core.library.types.QuantityType;
-import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
-import org.eclipse.smarthome.core.types.State;
-import org.eclipse.smarthome.core.types.UnDefType;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
 
 /**
  * this class is used to map the aggregate data response of the public API
  *
  * @author Alexander Friese - initial contribution
  */
-public class AggregateDataResponsePublicApi implements DataResponse {
-
-    private final Logger logger = LoggerFactory.getLogger(AggregateDataResponsePublicApi.class);
-
-    private static final String METER_TYPE_PRODUCTION = "Production";
-    private static final String METER_TYPE_CONSUMPTION = "Consumption";
-    private static final String METER_TYPE_SELFCONSUMPTION = "SelfConsumption";
-    private static final String METER_TYPE_IMPORT = "Purchased";
-    private static final String METER_TYPE_EXPORT = "FeedIn";
+@NonNullByDefault
+public class AggregateDataResponsePublicApi {
 
+    @NonNullByDefault
```

Comment:
Null annotations on inner classes are not necessary. There is a bug in checkstyle, generating a false positive warning, which can safely be ignored. Same for the others.

Reply:
I recognized that there are some differences between warnings shown by Eclipse and those shown by the mvn build process.
But I thought that Eclipse was wrong. I will remove these.